packages/react-sdk/src/components/connection-state/ConnectionStateProvider.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import WarningAmberIcon from '@mui/icons-material/WarningAmber';
+import { Box, Button, useTheme } from '@mui/material';
+import React, {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNetworkState } from 'react-use';
+import { useSnackbar } from '../../components/Snackbar';
+
+type ConnectionState = 'online' | 'no_internet_connection';
+
+export type ConnectionStateState = {
+  connectionState: ConnectionState;
+  connectionStateDialogOpen: boolean;
+  handleCloseConnectionStateDialog: () => void;
+};
+
+export const ConnectionStateContext = createContext<
+  ConnectionStateState | undefined
+>(undefined);
+
+export const ConnectionStateProvider: React.FC<PropsWithChildren> = function ({
+  children,
+}) {
+  const theme = useTheme();
+  const { t } = useTranslation();
+  const { clearSnackbar, showSnackbar, snackbarProps } = useSnackbar();
+  const networkState = useNetworkState();
```

Comment:
It might be worth adding a comment that in I believe all browsers this only tells you if the router is unreachable. However, a working LAN without Internet/Server connection would still be silently failing here. https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine mentions the behaviour which https://github.com/streamich/react-use/blob/e1d0cd9f7fb2a124a9d46879489abfefdf48d836/src/useNetworkState.ts#L102-L130 uses via the event listeners.

Function wise this might be ok but we should be aware of this behaviour when we look at this in $x months again.

Reply:
Jep, you are right. The browser network connection API is not great. I have also mentioned it in the ticket already. Next PR will be active monitoring by sending requests to the homeserver.

Comment added.